docs/guides/oomkill.md

```diff
@@ -0,0 +1,92 @@
+---
+title: 'The "ommkil" gadget'
+weight: 10
+---
+
+The `oomkill` gadget is used to monitor when OOM killer actually kills a process.
+In this guide, we will learn how to use it by running a small `kubernetes` cluster inside `minikube`.
+
+## How to use it?
+
+First, we need to create one pod which will allocate all the memory:
+```bash
+$ kubectl create ns test-ns
+$ kubectl run -n test-ns --image debian:latest test-pod sleep inf
+```
+
+You can now use the gadget, but output will be empty:
+```bash
+$ kubectl gadget oomkill
+NODE             NAMESPACE        POD              CONTAINER        KPID   KCOMM            PAGES  TPID             TCOMM
+```
+
+Indeed, it is waiting for OOM killer to kick in.
+So, in *an other terminal*, `exec` a container and run this `perl` one-liner to exhaust the memory:
+```bash
+$ kubectl get pod -n test-ns
+NAME       READY   STATUS    RESTARTS   AGE
+test-pod   1/1     Running   0          13s
+$ kubectl exec -n test-ns -ti test-pod -- perl -e 'for(;;) { $a .= "A" x 1024; }'
+command terminated with exit code 137
+```
+
+Go back to *the first terminal* and see:
+```bash
+NODE             NAMESPACE        POD              CONTAINER        KPID   KCOMM            PAGES  TPID             TCOMM
+minikube         test-ns          test-pod         test-pod         85282  perl             718153 2757   kube-controller
+```
+
+The printed lined correspond to the killing of `perl` process by the OOM killer.
+Here is the full legend of all the fields:
+
+* `KPID`: The PID of the process killed by the OOM killer (KilledPID).
+* `KOMM`: The command of the process killed by the OOM killer (KilledCommand).
+* `PAGES`: The number of pages the killed process had. A page is 4096 bytes on majority of operating system.
+* `TPID`: The PID of the process which triggered the OOM killer (TriggeredPID).
+* `TCOMM`: The command of the process which triggered the OOM killer (TriggeredCommand).
+
+So, the above line should be read like this: "Command perl, whom PID is 85282, running inside container pod, within pod pod, in default namespace on minikube node, was killed by the OOM killer because it allocated 908025 pages. The OOM killer was triggered by kube-controller whom pid is 2757."
+
+## Global output
+
+To be sure no events are missed, it is possible to use the `-A/--all-namespaces` to monitor all existing namespaces:
+
+```bash
+$ ./kubectl-gadget oomkill -A
+NODE             NAMESPACE        POD              CONTAINER        KPID   KCOMM            PAGES  TPID             TCOMM
+minikube         test-ns          test-pod         test-pod         87691  perl             718153 2479   kubelet
+```
```

Comment:
I don't understand this. Why are events missed if this option is not used?

Reply:
If you monitor for a given `namespace` and the OOM killer is called and kills a process in another namespace you will miss it.
The problem is that OOM killer events are, in my opinion, quite important so better to not miss them using `-A` option.

Reply:
I agree. However I'd remove that part of the guide as it doesn't add any useful information IMO. That option is supported by almost all the gadgets so I don't see why it needs a special mention here.

Reply:
Let's find a middle ground: I will remove the detailed part about it but I will still mention "somewhere" user can use the `-A` just in case the user just wants to use `oomkill` and is not familiar with all our gadgets.